mono/metadata/icall.c

```diff
@@ -160,28 +160,47 @@ mono_icall_get_file_path_prefix (const gchar *path)
 }
 #endif /* HOST_WIN32 */
 
-ICALL_EXPORT MonoObject *
-ves_icall_System_Array_GetValueImpl (MonoArray *arr, guint32 pos)
+ICALL_EXPORT MonoObjectHandle
+ves_icall_System_Array_GetValueImpl (MonoArrayHandle arr, guint32 pos, MonoError *error)
 {
-	ERROR_DECL (error);
 	MonoClass *ac;
 	gint32 esize;
 	gpointer *ea;
-	MonoObject *result = NULL;
+	guint gchandle = 0;
+	MonoObjectHandle result;
 
-	ac = (MonoClass *)arr->obj.vtable->klass;
+	ac = mono_handle_class (arr);
 
 	esize = mono_array_element_size (ac);
-	ea = (gpointer*)((char*)arr->vector + (pos * esize));
+	ea = (gpointer*)mono_array_handle_pin_with_size (arr, esize, pos, &gchandle);
+
+	if (ac->element_class->valuetype)
+#if 0	// FIXME merge with https://github.com/mono/mono/pull/6876
+	// FIXME merge with https://github.com/mono/mono/pull/6903
```

Comment:
Dead code: either remove or fix.

Reply:
Those other commits are in now so should resolve well.